### ISSUE TITLE
Docs specify incorrect memory order for Timestep 

### ISSUE THREAD

**Issue by hmacdope** [MDAnalysis/mdanalysis#3675]
## Expected behavior ##

The docs specify the memory order of `timestep._pos' correctly to be "C".

## Actual behavior ##
The docs [here](https://docs.mdanalysis.org/stable/documentation_pages/coordinates/base.html?highlight=_pos#MDAnalysis.coordinates.base.Timestep._pos)

specifies   
```
_pos  of dtype float32 of shape (n_atoms, 3) and internal FORTRAN order, holding the raw cartesian coordinates ... 

``` 
Where 'FORTRAN' should be replaced by 'C'

**Comment by orbeckst** [comment]
Can you add to the issue report what the correct answer is?

**Comment by hmacdope** [comment]
Done, apologies should have left more detail.

**Comment by hmacdope** [comment]
This was fixed in #3683
